src/Microsoft.CorrelationVector/CorrelationVectorV1.cs

```diff
@@ -0,0 +1,404 @@
+            }
+        }
+
+        private static bool IsImmutable(string correlationVector)
+        {
+            return !string.IsNullOrEmpty(correlationVector) && correlationVector.EndsWith(CorrelationVectorV1.TerminationSign);
+        }
+
+        private static bool IsOversized(string baseVector, int extension, CorrelationVectorVersion version)
+        {
+            if (!string.IsNullOrEmpty(baseVector))
+            {
+                int size = baseVector.Length + 1 +
+                    (extension > 0 ? (int)Math.Log10(extension) : 0) + 1;
+                return ((version == CorrelationVectorVersion.V1 &&
+                      size > CorrelationVectorV1.MaxVectorLength) ||
+                     (version == CorrelationVectorVersion.V2 &&
+                      size > CorrelationVectorV1.MaxVectorLengthV2));
+            }
+            return false;
+        }
+
+        private static void Validate(string correlationVector, CorrelationVectorVersion version)
+        {
+            byte maxVectorLength;
+            byte baseLength;
+
+            if (CorrelationVectorVersion.V1 == version)
+            {
+                maxVectorLength = CorrelationVectorV1.MaxVectorLength;
+                baseLength = CorrelationVectorV1.BaseLength;
+            }
+            else if (CorrelationVectorVersion.V2 == version)
+            {
+                maxVectorLength = CorrelationVectorV1.MaxVectorLengthV2;
+                baseLength = CorrelationVectorV1.BaseLengthV2;
+            }
+            else
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, "Unsupported correlation vector version: {0}", version));
+            }
+
+            if (string.IsNullOrWhiteSpace(correlationVector) || correlationVector.Length > maxVectorLength)
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture,
+                    "The {0} correlation vector can not be null or bigger than {1} characters", version, maxVectorLength));
+            }
+
+            string[] parts = correlationVector.Split('.');
+
+            if (parts.Length < 2 || parts[0].Length != baseLength)
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, "Invalid correlation vector {0}. Invalid base value {1}", correlationVector, parts[0]));
+            }
+
+            for (int i = 1; i < parts.Length; i++)
+            {
+                int result;
+                if (int.TryParse(parts[i], out result) == false || result < 0)
+                {
+                    throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, "Invalid correlation vector {0}. Invalid extension value {1}", correlationVector, parts[i]));
+                }
+            }
+        }
+
+        public override Tuple<string, string> Reset()
+        {
+            throw new NotImplementedException();
```

Comment:
We should throw another Exception with a message stating that this feature is not available for this certain cV version